README.md

```diff
@@ -26,4 +26,195 @@
+
+
+## Use cases
+
+In the current state, `srai` provides the following functionalities:
+
+* **OSM data download** - downloading OpenStreetMap data for a given area using different sources
+* **OSM data processing** - processing OSM data to extract useful information (e.g. road network, buildings, POIs, etc.)
+* **GTFS processing** - extracting features from GTFS data
+* **Regionization** - splitting a given area into smaller regions using different algorithms (e.g. Uber's H3[1], Voronoi, etc.)
+* **Embedding** - embedding regions into a vector space based on different spatial features, and using different algorithms (eg. hex2vec[2], etc.)
+* Utilities for spatial data visualization and processing
+
+For future releases, we plan to add more functionalities, such as:
+
+* **Pre-computed embeddings** - pre-computed embeddings for different regions and different embedding algorithms
+* **Full pipelines** - full pipelines for different embedding approaches, pre-configured from `srai` components
+* **Image data download and processing** - downloading and processing image data (eg. OSM tiles, etc.)
+
+
+## Installation
+
+To install `srai` simply run:
+
+```bash
+pip install srai
+```
+
+This will install the `srai` package and dependencies required by most of the use cases. There are several optional dependencies that can be installed to enable additional functionality. These are listed in the [optional dependencies](#optional-dependencies) section.
+
+### Optional dependencies
+
+The following optional dependencies can be installed to enable additional functionality:
+
+* `srai[all]` - all optional dependencies
+* `srai[osm]` - dependencies required to download OpenStreetMap data
+* `srai[voronoi]` - dependencies to use Voronoi-based regionization method
+* `srai[gtfs]` - dependencies to process GTFS data
+* `srai[plotting]` - dependencies to plot graphs and maps
+* `srai[torch]` - dependencies to use torch-based embedders
+
+## Usage
+
+### Downloading OSM data
+
+To download OSM data for a given area, using a set of tags use one of `OSMLoader` classes:
+
+* `OSMOnlineLoader` - downloads data from OpenStreetMap API - this is faster for smaller areas or tags counts
+* `OSMPbfLoader` - loads data from automatically downloaded PBF file - this is faster for larger areas or tags counts
+
+Example with `OSMPbfLoader`:
+
+```python
+from srai.loaders import OSMOnlineLoader
+from srai.utils import geocode_to_region_gdf
+
+filter = {"leisure": "park"}
+area = geocode_to_region_gdf("Wrocław, Poland")
+loader = OSMOnlineLoader()
+
+parks_gdf = loader.load(area, filter)

```

Comment:
We should probably change `filter` to some other name (`query`?) since filter is a keyword in python and syntax highlights it.

Reply:
Good point!